source/databases-collections.txt

```diff
@@ -0,0 +1,286 @@
+
+- **Databases**: Top-level data structures in a MongoDB deployment that store collections.
+- **Collections**: Groups of MongoDB documents. They are analogous to tables in relational databases.
+- **Documents**: Units that store literal data such as string, numbers, dates, and other embedded documents.
+  For more information about document field types and structure, see the
+  :manual:`Documents </core/document/>` guide in the {+mdb-server+} manual.
+
+Access a Database
+-----------------
+
+Access a database by passing the database name to the ``MongoDB\Client::selectDatabase()``
+method.
+
+The following example accesses a database named ``test_database``:
+
+.. literalinclude:: /includes/databases-collections/databases-collections.php
+    :language: php
+    :dedent:
+    :start-after: start-access-database
+    :end-before: end-access-database
+
+.. _php-db-coll-access-collection:
+
+Access a Collection
+-------------------
+
+Access a collection by using either of the following methods:
+
+- ``MongoDB\Client::selectCollection()``: Pass the database and collection names as 
+  parameters
+- ``MongoDB\Database::selectCollection()``: Pass the collection name as a parameter
+
+The following example accesses a collection named ``test_collection`` by using the 
+``MongoDB\Database::selectCollection()`` method:
+
+.. literalinclude:: /includes/databases-collections/databases-collections.php
+    :language: php
+    :dedent:
+    :start-after: start-access-collection
+    :end-before: end-access-collection
+
+.. tip::
+
+   If the provided collection name does not already exist in the database,
+   MongoDB implicitly creates the collection when you first insert data
+   into it.
+
+.. _php-db-coll-create-collection:
+
+Create a Collection
+-------------------
+
+Pass a collection name to the ``MongoDB\Database::createCollection()`` method to
+explicitly create a collection in a MongoDB database.
+
+The following example creates a collection named ``example_collection``:
+
+.. literalinclude:: /includes/databases-collections/databases-collections.php
+    :language: php
+    :dedent:
+    :start-after: start-create-collection
+    :end-before: end-create-collection
+
+You can specify collection options, such as maximum size and document
+validation rules, by passing them as an array to the ``createCollection()`` method.
+For a full list of optional parameters, see the `API documentation
+<{+api+}/method/MongoDBDatabase-createCollection/#parameters>`__.
+
+Get a List of Collections
+-------------------------
+
+You can query for a list of collections in a database by calling the
+``MongoDB\Database::listCollections()`` method. The method returns a
+cursor containing all collections in the database and their associated metadata.
+
+The following example calls the ``listCollections()`` method and iterates over
+the returned cursor to print the collections from the :ref:`php-db-coll-access-collection`
```

Comment:
```suggestion
the returned iterator to print the collections from the :ref:`php-db-coll-access-collection`
```

Although the [`listCollections` command](https://www.mongodb.com/docs/manual/reference/command/listCollections/) returns a cursor, PHPLIB abstracts that with another iterator. This allows us to do things like [cache the results](https://github.com/mongodb/mongo-php-library/blob/v1.x/src/Model/CachingIterator.php) and return typed model classes (e.g. CollectionInfo).

Caching is helpful if users want to iterate the result multiple times, as the original cursor can only be iterated once as it streams results from the server.

Also note that the special iterator classes returned by the database, collection, and index enumeration methods are being deprecated and will be removed in 2.0 (see: [PHPLIB-1522](https://jira.mongodb.org/browse/PHPLIB-1522)). The individual result classes (e.g. CollectionInfo) will remain, though. In order to future-proof the docs, I propose we just talk about these methods returning an iterator.